airbyte-webapp/src/views/Connector/ConnectorCard/ConnectorCard.tsx

```diff
@@ -1,31 +1,51 @@
-import React, { useEffect, useState } from "react";
+import React, { useEffect, useMemo, useState } from "react";
 import { FormattedMessage } from "react-intl";
 
 import { JobItem } from "components/JobItem/JobItem";
 import { Card } from "components/ui/Card";
 
-import { Connector, ConnectorSpecification, ConnectorT } from "core/domain/connector";
+import {
+  Connector,
+  ConnectorDefinition,
+  ConnectorDefinitionSpecification,
+  ConnectorSpecification,
+  ConnectorT,
+} from "core/domain/connector";
 import { SynchronousJobRead } from "core/request/AirbyteClient";
 import { LogsRequestError } from "core/request/LogsRequestError";
 import { useAdvancedModeSetting } from "hooks/services/useAdvancedModeSetting";
 import { generateMessageFromError } from "utils/errorStatusMessage";
-import { ServiceForm, ServiceFormProps, ServiceFormValues } from "views/Connector/ServiceForm";
+import { ConnectorFormValues, ServiceForm, ServiceFormValues } from "views/Connector/ServiceForm";
 
+import { useDocumentationPanelContext } from "../ConnectorDocumentationLayout/DocumentationPanelContext";
 import { ConnectorServiceTypeControl } from "../ServiceForm/components/Controls/ConnectorServiceTypeControl";
 import styles from "./ConnectorCard.module.scss";
 import { useAnalyticsTrackFunctions } from "./useAnalyticsTrackFunctions";
 import { useTestConnector } from "./useTestConnector";
 
-type ConnectorCardProvidedProps = Omit<
-  ServiceFormProps,
-  "isKeyConnectionInProgress" | "isSuccess" | "onStopTesting" | "testConnector"
->;
-
-interface ConnectorCardBaseProps extends ConnectorCardProvidedProps {
+// TODO: need to clean up the ConnectorCard and ServiceForm props,
+// since some of props are used in both components, and some of them used just as a prop-drill
```

Comment:
Could you create an issue for this, which captures why the current state is bad and what might be a better solution?

Reply:
Sure, here is it: https://github.com/airbytehq/airbyte/issues/18553
I will add it to comment also